test/suite/helper_tiered.py

```diff
@@ -90,3 +90,59 @@ def generate_s3_prefix(test_name = ''):
 
     return prefix
 
+# Storage sources
+tiered_storage_sources = [
+    ('dirstore', dict(auth_token = get_auth_token('dir_store'),
+        bucket = get_bucket1_name('dir_store'),
+        bucket_prefix = "pfx_",
+        ss_name = 'dir_store')),
+    ('s3', dict(auth_token = get_auth_token('s3_store'),
+        bucket = get_bucket1_name('s3_store'),
+        bucket_prefix = generate_s3_prefix(),
+        ss_name = 's3_store')),
+    ('non_tiered', dict(ss_name = 'non_tiered')),            
+]
+
+# Retruns True if the current scenario is tiered
+def is_tiered_scenario(test_case):
+    res = False
+    if hasattr(test_case, 'ss_name'):
+        res = test_case.ss_name != 'non_tiered'
+    
+    return res
+
+# Setup custom connection config.
+def tiered_conn_config(test_case):
+    # Handle non_tiered storage scenarios.
+    if not is_tiered_scenario(test_case):
+        return ''
+
+    # Setup directories structure for local tiered storage.
+    if test_case.ss_name == 'dir_store' and not os.path.exists(test_case.bucket):
```

Comment:
Again, avoid comparisons against `ss_name`, but rather have a field set by the scenario.   Could be either a boolean, or maybe in this case maybe `self.pre_create_directory`.

I know other tests use the pattern to compare against a name, but it isn't necessarily good practice.

Reply:
I agree. I added `is_local_storage` flag to the disctionary.